syntax = "proto3";

import "google/protobuf/any.proto";

package heph.ipc.zenoh.action_server.proto;

enum RequestStatus {
    SUCCESSFUL = 0;
    REJECTED_USER = 1;
    REJECTED_ALREADY_RUNNING = 2;
    INVALID = 3;
    STOPPED = 4;
}

message RequestResponse {
    RequestStatus status = 1;
};

message Response {
    RequestStatus status = 1;
    google.protobuf.Any value = 2;
}
